src/foam/u2/grid/GridCell.js

```diff
@@ -0,0 +1,349 @@
+                d.select().
+                then(function(result){
+                    var div = foam.u2.Element.create("div");
+                    console.log("CELL: row:" + this.rowMatchId + " col:" + this.colMatchId + ", " + result.a.length);
+                    if (! result || !result.a || !result.a.length){
+                        console.log("no result found");
+                    }
+                    var a = result.a;
+                    for (var i=0; i<a.length; i++){
+                        var entry = a[i];
+                        var v = this.getEntryView(entry); 
+                        v.sub("CELL_ENTRY_SELECTED", this.onEntrySelection)
+                        div.add(v); 
+                    }
+                    div.add(this.cellBottom); 
+                    this.cell = div;
+                }.bind(this));
+            }else {
+                var p = foam.u2.Element.create("div");
+                p.add("---");
+                this.cell = p;
+            }
+        },
+        
+        function getEntryView(entry){
+            console.log(entry);
+            var v = this.getCellView(entry);
+            if (self.entrySelectable){
+                v.sub("SELECTED", function(e){
+                    self.selectedEntry = e.src.data; 
+                    self.pub("ENTRY_SELECTION");
+                    }.bind(this));
+            }
+            return v; 
+        }, 
+        
+        function getCellView(a){
+            if (this.cellView){
+                var v = this.cellView$cls.create({of: this.of, data: a});
+                return v; 
+            }
+            var d = foam.u2.Element.create("div");
+            d.add(foam.u2.Element.create("p").add(a.name));
+            d.add(foam.u2.Element.create("p").add(a.lastSeenAlive));
+            return d; 
+        }, 
+        
+        function toggleRowHighlight(){
+            this.inSelectedRow = ! this.inSelectedRow;
+        },
+        
+        
+        function toggleColHighlight(){
+            this.inSelectedCol = ! this.inSelectedCol;
+        },
+        
+        function clearCSS(){
+        }, 
+        
+        
+    ],
+    
+    listeners: [
+        {
+            name: 'onClick',
+            isFramed: true,
+            code: function(){
+                this.selected = !this.selected;
+                this.pub("CELL_CLICK");
+            }
+                
+        },
+        
+        {
+            name: 'onEntrySelection',
+            isFramed: true,
+            code: function(a, b, c){
+                this.pub("ENTRY_SELECTION");
+                console.log("entry selected in GridCel.js");
+                if (a.src && a.src.data){
+                    this.selectedEntry = a.src.data;
+                    this.pub("ENTRY_SELECTION"); 
+                }
+            }
+                
+        },
+        
+        {
+            name: 'onDataUpdate',
+            isFramed: true,
+            mergeDelay: 1000, 
```

Comment:
Hi, Kevin: 

This is roughly what I have. 

Thank you for reviewing, and please tell me what I lack and any design stupidities I've made. 

I've tried to do my best to implement this, but it feels rather clunky to me. 
In the pull request, I have added an example that runs well. 
(there are some other examples that combines the usage of GridView and FilterController, but I need to fix them first. )

Also, I'll fix everything to be foam perfect. 
Thank you for you help. I'm looking forward learning from you. 
Yang